### ISSUE TITLE
[3.4b4] Commas in grouping values break the listings

### ISSUE THREAD

**Issue by xiaohutai** [bolt/bolt#7107]
Related to #7096 

If  you have `grouping` taxonomy with a comma `,` in the value, the back-end overview breaks, resulting in:

```
Key "" for array with keys "tags, groups, categories" does not exist in "@bolt/_base/_listing.twig" at line 315.
```

Details
-------

 - Relevant Bolt Version: [ 3.4-beta4 ]
 - Install type: [ Zip/tarball install | Composer install ]
 - PHP version: [ 5.6]
 - Used web server: [ Apache ]

Reproduction
------------

* Install Bolt
* Modify `taxonomy.yml` to add a `,` in the value part of the value

```diff
groups:
    slug: groups
    singular_slug: group
    behaves_like: grouping
-    options: { main: "The main group", meta: "Meta group", other: "The other stuff" }
+    options: { main: "The, main group", meta: "Meta, group", other: "The, other stuff" }
    has_sortorder: true
```
* Prefill Bolt with some Pages

Edit
------------
- Note: It did work for me on Bolt 3.3.6

**Comment by GawainLynch** [comment]
The problem is emanating [from here](https://github.com/bolt/bolt/blob/3.4/src/Storage/Field/Type/JoinTypeBase.php#L51-L53)

ping @rossriley

**Comment by GawainLynch** [comment]
Fixed in #7130
